cmd/serverless-init/initcontainer/initcontainer.go

```diff
@@ -0,0 +1,116 @@
+package initcontainer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/serverless-init/log"
+	"github.com/DataDog/datadog-agent/cmd/serverless-init/metric"
+	"github.com/DataDog/datadog-agent/cmd/serverless-init/tag"
+	"github.com/DataDog/datadog-agent/pkg/logs"
+	"github.com/DataDog/datadog-agent/pkg/serverless/metrics"
+	"github.com/DataDog/datadog-agent/pkg/serverless/trace"
+)
+
+func Run(logConfig *log.Config, metricAgent *metrics.ServerlessMetricAgent, traceAgent *trace.ServerlessTraceAgent, args []string) {
+	log.Write(logConfig, []byte(fmt.Sprintf("[datadog init process] running cmd = >%v<", args)))
+	err := execute(logConfig, metricAgent, traceAgent, args)
+	if err != nil {
+		log.Write(logConfig, []byte(fmt.Sprintf("[datadog init process] exiting with code = %s", err)))
+	} else {
+		log.Write(logConfig, []byte("[datadog init process] exiting successfully"))
+	}
+}
+
+func execute(config *log.Config, metricAgent *metrics.ServerlessMetricAgent, traceAgent *trace.ServerlessTraceAgent, args []string) error {
```

Comment:
Will the ServerlessMetricsAgent and ServerlessTraceAgent, logs periodically flush?  I thought we disabled that behaviour.

Reply:
- cold start is metric is flushed right after being emitted
- all other metrics are flushed when the container is killed
- traces are flushed every 3s for now

we could change this behaviour of course but we might want to try that for the private beta?

Reply:
So any custom metrics might not be flushed for a long time?